test/test-mc-schema-broker.c

```diff
@@ -40,7 +40,9 @@ static void test_mc_schema_broker_request(_mongocrypt_tester_t *tester) {
         mongocrypt_status_t *status = mongocrypt_status_new();
         mc_schema_broker_t *sb = mc_schema_broker_new();
         ASSERT_OK_STATUS(mc_schema_broker_request(sb, "db", "coll1", status), status);
+        ASSERT(!mc_schema_broker_has_multiple_requests(sb));
```

Comment:
Suggest also asserting the `len == 0` case prior to the first call to `mc_schema_broker_request` both here (0 -> 1 -> 2) and below (0 -> 1 -> 1 (duplicate)).